### PR TITLE
[#9419] Add form validation for Instructor Course Student Edit Page

### PR DESCRIPTION
**Pull request by tanhengyeow**
Part of #9419 

Common checks are implemented. More complex checks would be delegated to the back-end.

### CODE REVIEW

File: src/web/app/pages-instructor/instructor-course-student-edit-page/instructor-course-student-edit-page.component.ts [hunk 1]
```diff
@@ -128,6 +136,23 @@ export class InstructorCourseStudentEditPageComponent implements OnInit, OnDestr
             .subscribe(() => this.isEmailFieldChanged = true);
   }
 
+  /**
+   * Displays message to user stating that the field is empty.
+   */
+  displayEmptyFieldMessage(fieldName: string, maxLength: number): string {
+    return `The field '${fieldName}' is empty.` +
```
Comment (monmanuela) [inline src/web/app/pages-instructor/instructor-course-student-edit-page/instructor-course-student-edit-page.component.ts]: I think `'${fieldName} should not be empty.` or `Please enter a ${fieldName}.` would be sufficient.
My reasoning is that when the field is empty, the user doesn't need to care about the `maxLength` (yet) until he actually exceeds the max length, which will trigger `displayExceedMaxLengthMessage`. A shorter message would also reduce cognitive load. What do you think?

File: src/web/app/pages-instructor/instructor-course-student-edit-page/instructor-course-student-edit-page.component.ts [hunk 2]
```diff
@@ -128,6 +136,23 @@ export class InstructorCourseStudentEditPageComponent implements OnInit, OnDestr
...
+        ` The value of a/an ${fieldName} should be no longer than ${maxLength} characters.` +
+        ' It should not be empty.';
+  }
+
+  /**
+   * Displays message to user stating that the field exceeds the max length.
+   */
+  displayExceedMaxLengthMessage(fieldName: string, maxLength: number): string {
+    return `The input is not acceptable to TEAMMATES as a/an '${fieldName}'` +
```
Comment (monmanuela) [inline src/web/app/pages-instructor/instructor-course-student-edit-page/instructor-course-student-edit-page.component.ts]: Perhaps `${fieldName} should not exceed ${maxLength} characters.` would be better?

File: src/web/app/pages-instructor/instructor-course-student-edit-page/instructor-course-student-edit-page.component.ts (hunk 1 above)
Comment (tanhengyeow) [reply src/web/app/pages-instructor/instructor-course-student-edit-page/instructor-course-student-edit-page.component.ts]: @monmanuela That's a valid point. Most of the messages included here were referenced from `FieldValidator.java` (back-end validation). 

A shorter and more succinct message would be more suitable for front-end validation. We might want to keep this in mind and improve the back-end validation message if appropriate.

File: src/web/app/pages-instructor/instructor-course-student-edit-page/instructor-course-student-edit-page.component.ts (hunk 2 above)
Comment (tanhengyeow) [reply src/web/app/pages-instructor/instructor-course-student-edit-page/instructor-course-student-edit-page.component.ts]: Same comment as https://github.com/TEAMMATES/teammates/pull/9633#discussion_r272031420

**Comment by tanhengyeow** [comment]
New look/message:
image

Used hexcode `#B50000` to maintain a color contrast ratio of > 5, might want to keep this in mind for other pages with form validation of different background color:
image

**Pull request merged**